hbase/hbase/stackable/patches/2.6.1/0006-Update-property-usage-for-bound-ports.patch

```diff
@@ -0,0 +1,174 @@
+-    int port = conf.getInt(REGIONSERVER_BOUND_INFO_PORT, 0);
++    int port = conf.getInt(REGIONSERVER_INFO_PORT, 0);
+     return port != 0 ? port : this.infoServer != null ? this.infoServer.getPort() : -1;
+   }
+ 
+@@ -2463,11 +2464,11 @@ public class HRegionServer extends Thread
+    */
+   private void putUpWebUI() throws IOException {
+     int port =
+-      this.conf.getInt(HConstants.REGIONSERVER_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT);
++      this.conf.getInt(REGIONSERVER_BOUND_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT);
+     String addr = this.conf.get("hbase.regionserver.info.bindAddress", "0.0.0.0");
+ 
+     if (this instanceof HMaster) {
+-      port = conf.getInt(HConstants.MASTER_INFO_PORT, HConstants.DEFAULT_MASTER_INFOPORT);
++      port = conf.getInt(HConstants.MASTER_BOUND_INFO_PORT, HConstants.DEFAULT_MASTER_INFOPORT);
+       addr = this.conf.get("hbase.master.info.bindAddress", "0.0.0.0");
+     }
+     // -1 is for disabling info server
+@@ -2503,12 +2504,18 @@ public class HRegionServer extends Thread
+         LOG.info("Retry starting http info server with port: " + port);
+       }
+     }
+-    port = useThisInfoPortInstead;
+-    conf.setInt(HConstants.REGIONSERVER_INFO_PORT, port);
++
++    // update bound ports
++    port = this.infoServer.getPort();
++    conf.setInt(REGIONSERVER_BOUND_INFO_PORT, port);
++    conf.setInt(HConstants.MASTER_BOUND_INFO_PORT, port);
++
++    // set advertised ports
++    conf.setInt(REGIONSERVER_INFO_PORT, useThisInfoPortInstead);
+     int masterInfoPort =
+       conf.getInt(HConstants.MASTER_INFO_PORT, HConstants.DEFAULT_MASTER_INFOPORT);
+     conf.setInt("hbase.master.info.port.orig", masterInfoPort);
+-    conf.setInt(HConstants.MASTER_INFO_PORT, port);
++    conf.setInt(HConstants.MASTER_INFO_PORT, useThisInfoPortInstead);
+   }
+ 
+   /*
+diff --git a/hbase-server/src/test/java/org/apache/hadoop/hbase/util/ProcessBasedLocalHBaseCluster.java b/hbase-server/src/test/java/org/apache/hadoop/hbase/util/ProcessBasedLocalHBaseCluster.java
+index 21c49e61ba..2dd85ab764 100644
+--- a/hbase-server/src/test/java/org/apache/hadoop/hbase/util/ProcessBasedLocalHBaseCluster.java
++++ b/hbase-server/src/test/java/org/apache/hadoop/hbase/util/ProcessBasedLocalHBaseCluster.java
+@@ -370,13 +370,13 @@ public class ProcessBasedLocalHBaseCluster {
+ 
+       int masterInfoPort = HBaseTestingUtility.randomFreePort();
+       reportWebUIPort("master", masterInfoPort);
+-      confMap.put(HConstants.MASTER_INFO_PORT, masterInfoPort);
++      confMap.put(HConstants.MASTER_BOUND_INFO_PORT, masterInfoPort);
+     } else if (serverType == ServerType.RS) {
+       confMap.put(HConstants.REGIONSERVER_PORT, rpcPort);
+ 
+       int rsInfoPort = HBaseTestingUtility.randomFreePort();
+       reportWebUIPort("region server", rsInfoPort);
+-      confMap.put(HConstants.REGIONSERVER_INFO_PORT, rsInfoPort);
++      confMap.put(HConstants.REGIONSERVER_BOUND_INFO_PORT, rsInfoPort);
```

Comment:
Assuming that we do the fallback, existing tests should be ok as is, I think?

Reply:
That was changed inadvertently - have reverted it: 65aaae2